components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -2699,6 +2708,9 @@ protected void createAPI(API api) throws APIManagementException {
             String[] visibleRoles = new String[0];
             if (visibleRolesList != null) {
                 visibleRoles = visibleRolesList.split(",");
+                for(int i = 0; i < visibleRoles.length; i++) {
```

Comment:
Formatting issues

Reply:
Lets also use enhanced for loop(https://stackoverflow.com/a/18738765/3296350), its a lot cleaner